pkg/metrics/metrics.go

```diff
@@ -59,6 +65,67 @@ const (
 	CQStatusTerminating ClusterQueueStatus = "terminating"
 )
 
+type LocalQueueMetricsConfig struct {
+	Enabled            bool
+	NamespaceSelector  *metav1.LabelSelector
+	LocalQueueSelector *metav1.LabelSelector
+}
+
+var lqMetricsConfigSingleton *LocalQueueMetricsConfig
+
+func SetLocalQueueMetrics(lqMetricsConfig *LocalQueueMetricsConfig) {
+	lqMetricsConfigSingleton = lqMetricsConfig
+}
+
+var once sync.Once
+
+func GetLocalQueueMetrics() *LocalQueueMetricsConfig {
+	if lqMetricsConfigSingleton == nil {
+		once.Do(
+			func() {
+				lqMetricsConfigSingleton = &LocalQueueMetricsConfig{}
+			})
+	}
+	return lqMetricsConfigSingleton
+}
```

Comment:
Why do we need this? Do you try to detect Configuration change?

Reply:
Just to make it clear that access to and changes to the singleton should be done via the two public functions provided rather than interacting directly with the singleton object